src/app/forecasting/input/ynab/ynab.component.ts

```diff
@@ -325,51 +384,87 @@ export class YnabComponent implements OnInit {
   }
 
   private mapAccounts(accounts: ynab.Account[]) {
-    const mapped = accounts.filter(a => !(a.closed || a.deleted))
-      .map(a => this.formBuilder.group(Object.assign({}, a, {
-        balance: this.getAccountBalance(a),
-        ynabBalance: ynab.utils.convertMilliUnitsToCurrencyAmount(a.balance)
-      })));
+    const mapped = accounts
+      .filter((a) => !(a.closed || a.deleted))
+      .map((account) => {
+        const ynabBalance = ynab.utils.convertMilliUnitsToCurrencyAmount(
+          account.balance
+        );
+        const overrides = NoteUtility.getNoteOverrides(
+          account.note,
+          ynabBalance
+        );
+
+        return this.formBuilder.group(
+          Object.assign({}, account, {
+            balance: this.getAccountBalance(account, ynabBalance, overrides),
+            ynabBalance,
+            monthlyContribution: overrides.monthlyContribution,
+          })
+        );
+      });
     return mapped;
   }
 
-  private getAccountBalance(account: ynab.Account) {
-    const balance = ynab.utils.convertMilliUnitsToCurrencyAmount(account.balance);
-    const overrides = NoteUtility.getNoteOverrides(account.note, balance);
-
+  private getAccountBalance(
+    account: ynab.Account,
+    ynabBalance: number,
+    overrides: Overrides
+  ) {
     if (overrides.contributionBudget !== undefined) {
       return overrides.contributionBudget;
     }
 
-    if (account.type === ynab.Account.TypeEnum.InvestmentAccount || account.type === ynab.Account.TypeEnum.OtherAsset) {
-      return balance;
+    if (
+      account.type === ynab.Account.TypeEnum.InvestmentAccount ||
+      account.type === ynab.Account.TypeEnum.OtherAsset
+    ) {
+      return ynabBalance;
     }
 
     return 0;
   }
 
-  private getMonthlyContribution(categoryGroups) {
+  private getMonthlyContribution(categoryGroups, accounts) {
     let contribution = 0;
     const categories = [];
 
     if (categoryGroups) {
-      categoryGroups.forEach(cg => {
-        cg.categories.forEach(c => {
+      categoryGroups.forEach((cg) => {
+        cg.categories.forEach((c) => {
           if (c.contributionBudget) {
             contribution += c.contributionBudget;
             categories.push({
               name: c.name,
               value: c.contributionBudget,
-              info: c.info
+              info: c.info,
             });
           }
         });
       });
     }
 
+    if (accounts) {
+      accounts.forEach((a) => {
+        const monthlyContribution = a.value.monthlyContribution;
+        if (monthlyContribution) {
+          contribution += monthlyContribution;
+          categories.push({
+            name: a.value.name,
+            value: monthlyContribution,
+            info: {
+              max: monthlyContribution,
+              mean: monthlyContribution,
+              min: monthlyContribution,
+            },
+          });
+        }

```

Comment:
This is pretty much the primary change and then just a little refactoring to get the monthlyContribution set up.